state/integration_test.go

```diff
@@ -0,0 +1,119 @@
+package state
+
+import (
+	"context"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/testutil/testnode"
+	blobtypes "github.com/celestiaorg/celestia-app/x/payment/types"
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+func TestIntegrationTestSuite(t *testing.T) {
+	suite.Run(t, new(IntegrationTestSuite))
+}
```

Comment:
Thank you! ❤️ 
Finally, we have tests for the state interactions. We should also allocate time after incentivized testnet to cover all other APIs with roundtrip tests(potential in the Swamp) Added this to our post-incentivized TODO list @renaynay